src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -251,6 +288,34 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
   }
 
   /**
+   * This is public for testing.
+   * This assumes that the value is not a large value that would be segmented by large message support.
+   */
+  public static ProducerRecord<byte[], byte[]> serializeWithHeaders(ExtensibleProducerRecord<byte[], byte[]> xRecord,
+    HeaderSerializerDeserializer headerParser) {
+
+    if (!xRecord.hasHeaders()) {
+      return new ProducerRecord<byte[], byte[]>(xRecord.topic(), xRecord.partition(), xRecord.timestamp(), xRecord.key(), xRecord.value());
+    }
+
+    int headersSize = headerParser.serializedHeaderSize(xRecord.headers());
+    int serializedValueSize = (xRecord.value() == null ? 0 : xRecord.value().length) + headersSize;
```

Comment:
This is actually the size without headers so I'm going to keep this.  I've removed all the code that tracks the header size in the consumer, but forgot to remove that same code in the producer so I will now remove that code as well.